### DISCUSSION TITLE
ARM64 uno-check requirement

### DISCUSSION THREAD

**Discussion by bstrother1** [unoplatform/uno#9119] [Q&A]
Been meaning to make this discussion, when installing uno-check on an ARM64 environment it looks at the wrong folder.  It looks at
DOTNET_ROOT which is set by default to /usr/local/share/dotnet/dotnet, which is a file not a directory. To get uno-check functioning DOTNET_ROOT has to be changed to usr/local/share/dotnet/x64/host/fxr/6.0.2 (or whatever dotnet your using). Trying ARM64 dotnet with uno-check outputs: 

Failed to load /usr/local/share/dotnet/host/fxr/6.0.2/libhostfxr.dylib, error: dlopen(/usr/local/share/dotnet/host/fxr/6.0.2/libhostfxr.dylib, 0x0001): tried: '/usr/local/share/dotnet/host/fxr/6.0.2/libhostfxr.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/usr/local/lib/libhostfxr.dylib' (no such file), '/usr/lib/libhostfxr.dylib' (no such file)

Also, why does Uno still use a checker when MAUI stopped supporting it? https://github.com/dotnet/maui/issues/5651

**Comment by bstrother1** [comment]
Should add uno-check is specifically looking for libhostfxr.dylib but not just that but an 'x86_64' version of it.

**Reply by jeromelaban** [reply]
Uno continues to use uno-check because there are additional checks we're adding, and it's also used to run on CI and setup the build properly. As for the rest, It would look like the DOTNET_ROOT value is incorrect, as it's supposed to point to a folder, not a file.

If you can provide the full uno-check log, it'll help.

**Reply by bstrother1** [reply]
Where do I look/what is the command to produce a uno-check log?

**Reply by jeromelaban** [reply]
You can find the options here: https://platform.uno/docs/articles/external/uno.check/doc/using-uno-check.html#troubleshooting

**Reply by bstrother1** [reply]
Will I need to wipe out my uno-check environment and start a fresh attempt at installation?
